### ISSUE TITLE
statistics view

### ISSUE THREAD

**Issue by kosarko** [ufal/dspace-angular#9]
v5 displays custom statistics (item views, total and individual bitstream download counts) per item, with drill down from all time -> individual year -> individual month.

The current (v5) implementation is in https://github.com/ufal/piwik-charts and has a live demo at https://ufal.github.io/piwik-charts/ (https://ufal.github.io/piwik-charts/single-chart-example.html or https://ufal.github.io/piwik-charts/multi-chart-example.html)

The examples reveal most of the api. The endpoint is `https://lindat.mff.cuni.cz/statistics/handle?h=<item_handle>&period=[year|month|day]&date=[yyyy|yyyy-MM]`

- `h` is the handle of an item, you get that from metadata field `dc.identifier.uri`, it's only ther `prefix/suffix` part (ie. you need to strip `http(s)://hdl.handle.net/` if present. This parameter is mandatory
- `date` - if missing it returns the overall + year based counts; if it is in the `yyyy` form (e.g. `2020`), it returns totals for the given year and a breakdown into individual months; if in the `yyyy-MM` (e.g. `2020-10`) it returns total for the given month + a breakdown into individual dates. It handles the month with and without the leading zero (`02` and `2`)
- ~~`period`~~ can be ignored (it's determinated by the date)

The implementation at https://github.com/ufal/piwik-charts/blob/master/js/piwik_charts.js does some sort of caching (if you zoom in a level and then, zoom out...). If it's a recent item such as `11234/1-5502` which has data only for the current year, it zooms in automatically ...

**Comment by kosarko** [comment]
there's some progress at https://github.com/kosarko/ng-piwik-stats

**Comment by amadulhaxxani** [comment]
@kosarko 
I found this in the app, no idea what it is. Looks similar to what I was creating but noticed it at the end.
You can try accessing this path:
/items/84d62f0e-d974-412a-8d75-422066950c5e/matomo-statistics

**Comment by kosarko** [comment]
> [@kosarko](https://github.com/kosarko) I found this in the app, no idea what it is. Looks similar to what I was creating but noticed it at the end. You can try accessing this path: /items/84d62f0e-d974-412a-8d75-422066950c5e/matomo-statistics

Yes, this was originally by @milanmajchrak, if memory servers well we've discussed this before starting this issue and come to the conclusion that this was never finished. I'll create an issue to clean this up.

**Comment by kosarko** [comment]
Doc:
for production setup we need
```
statistics:
  baseUrl: 'https://lindat.mff.cuni.cz/statistics'
  endpoint: "/handle"
```
in config.prod.yml
